deployments/puppet/Dockerfile

```diff
@@ -1,31 +1,27 @@
-FROM ubuntu:16.04
+FROM ruby:2.6-buster
 
 ENV PATH=$PATH:/opt/puppetlabs/bin:/opt/puppetlabs/pdk/bin
 
 WORKDIR /tmp
-RUN apt update &&\
-    apt install -y make wget vim gcc ruby ruby-dev libxml2 libxml2-dev libxslt1-dev git apt-transport-https ca-certificates g++
 
-RUN wget https://apt.puppetlabs.com/puppet6-release-xenial.deb &&\
-    dpkg -i puppet6-release-xenial.deb &&\
+RUN wget https://apt.puppetlabs.com/puppet6-release-buster.deb &&\
+    dpkg -i puppet6-release-buster.deb &&\
     rm *.deb &&\
     apt update &&\
     apt install -y puppet-agent
 
-RUN wget https://apt.puppet.com/puppet-tools-release-xenial.deb && \
-    dpkg -i puppet-tools-release-xenial.deb && \
+RUN wget https://apt.puppet.com/puppet-tools-release-buster.deb && \
+    dpkg -i puppet-tools-release-buster.deb && \
     apt-get update && \
     apt-get install -y pdk && \
     rm -f *.deb
 
 WORKDIR /etc/puppetlabs/code/modules/splunk_otel_collector
-COPY ./Rakefile ./Gemfile ./Gemfile.lock ./
+COPY ./ ./
 RUN gem install bundler && bundle install
 
 RUN mkdir -p /root/.config/puppet && \
     echo "---\n\
 disabled: true" > /root/.config/puppet/analytics.yml
 
 ENV PATH=/opt/puppetlabs/bin:$PATH
-
-COPY ./ ./
```

Comment:
This is unrelated to the automation, right? Can we merge it before this PR?

Reply:
These changes are required for blacksmith and the existing integration tests.